frame/mosaic-vault/src/lib.rs

```diff
@@ -0,0 +1,904 @@
+            
+			<MinTransferDelay<T>>::put(new_min_transfer_delay);
+
+			Self::deposit_event(Event::MinTransferDelayChanged{new_min_transfer_delay});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn set_max_fee(origin: OriginFor<T>, max_fee: T::Balance) -> DispatchResultWithPostInfo {
+			
+			T::AdminOrigin::ensure_origin(origin)?;
+            
+			ensure!(max_fee < T::FeeFactor::get(), Error::<T>::MaxFeeAboveFeeFactor);
+
+			ensure!(max_fee > Self::min_fee(), Error::<T>::MaxFeeBelowMinFee);
+
+            <MaxFee<T>>::put(max_fee);
+
+			Self::deposit_event(Event::MaxFeeChanged{max_fee});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn set_min_fee(origin: OriginFor<T>, min_fee: T::Balance) -> DispatchResultWithPostInfo {
+			
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(min_fee < Self::max_fee(), Error::<T>::MinFeeAboveMaxFee);
+            
+			ensure!(min_fee < T::FeeFactor::get(), Error::<T>::MinFeeAboveFeeFactor);
+
+            <MinFee<T>>::put(min_fee);
+
+			Self::deposit_event(Event::MinFeeChanged{min_fee});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn set_thresh_hold(origin: OriginFor<T>, new_fee_threshold: T::Balance) -> DispatchResultWithPostInfo {
+			 
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(new_fee_threshold < T::ThresholdFactor::get(), Error::<T>::ThresholdFeeAboveThresholdFactor);
+
+			<FeeThreshold<T>>::put(new_fee_threshold);
+
+			Self::deposit_event(Event::FeeThresholdChanged{new_fee_threshold});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn deposit(
+			 origin: OriginFor<T>, 
+			 amount: T::Balance,
+			 asset_id: T::AssetId, 
+			 destination_address: T::AccountId, 
+			 remote_network_id: T::RemoteNetworkId,
+		 	 transfer_delay: T::TransferDelay,
+			) -> DispatchResultWithPostInfo {
+
+			let sender = ensure_signed(origin)?;
+
+			ensure!(Self::pause_status() == false, Error::<T>::ContractPaused);
+
+			ensure!(amount != T::Balance::zero(), Error::<T>::ZeroAmount);
+
+			Self::only_supported_remote_token(remote_network_id.clone(), asset_id.clone())?;
+
+			ensure!(Self::last_transfer(&sender).checked_add(Self::transfer_lockup_time()).ok_or(Error::<T>::Overflow)? < T::BlockTimestamp::now().as_secs(), Error::<T>::TransferNotPossible);
+           
+			ensure!(transfer_delay >= <MinTransferDelay<T>>::get(), Error::<T>::TransferDelayBelowMinimum);
+		
+			ensure!(transfer_delay <= <MaxTransferDelay<T>>::get(), Error::<T>::TransferDelayAboveMaximum);
+
+			ensure!(amount <= Self::max_asset_transfer_size(asset_id), Error::<T>::AmountAboveMaxAssetTransferSize);
+
+			ensure!(amount >= Self::min_asset_transfer_size(asset_id), Error::<T>::AmountBelowMinAssetTransferSize);
+			// 
+			let pallet_account_id = Self::account_id();            
+            // move funds to pallet amount
+		     T::Currency::burn_from(asset_id, &sender, amount).map_err(|_|Error::<T>::BurnFromFailed)?;
```

Comment:
move mutating operations to the bottom, cause these can be partially applied and fail other stuff. so check first and mutate later. not sure, could make this `trnsactional` attribute as other option.